docs/docs/sourcing-from-prismic.md

```diff
@@ -68,160 +60,142 @@ require("dotenv").config({
+4. Create your page templates
 
-```js:title=gatsby-node.js
-const path = require("path")
+### Single Type
 
-exports.createPages = async ({ graphql, actions }) => {
-  const { createPage } = actions
+Because you defined a custom type as `Homepage` the query after the `prismic` key is called `allHomepages` or just `homepage`. You can also see the API IDs (from the field names) you created earlier. Your Homepage template could look something like this:
 
-  const pages = await graphql(`
-    {
-      allPrismicPost {
-        edges {
-          node {
-            id
-            uid
-          }
-        }
-      }
-    }
-  `)
-
-  const template = path.resolve("src/templates/post.jsx")
-
-  pages.data.allPrismicPost.edges.forEach(edge => {
-    createPage({
-      path: `/${edge.node.uid}`,
-      component: template,
-      context: {
-        uid: edge.node.uid,
-      },
-    })
-  })
-}
-```
+```jsx:title=src/templates/home.js
 
-```jsx:title=src/templates/post.jsx
 import React from "react"
 import { graphql } from "gatsby"
+import { RichText } from "prismic-reactjs"
+
+export const query = graphql`
+ query($lang: String!, $uid:String!){
+  prismic{
+    homepage(lang:$lang, uid:$uid){
+      title
+      content
+    }
+  }
+ }
+`
+const Homepage = props => {
+  const document = props.data.prismic.homepage
+  if (!document) return null
 
-const Post = ({ data: { prismicPost } }) => {
-  const { data } = prismicPost
   return (
-    <React.Fragment>
-      <h1>{data.title.text}</h1>
-      <div dangerouslySetInnerHTML={{ __html: data.content.html }} />
-    </React.Fragment>
+    <div>
+      <RichText render={document.title} />
+      <RichText render={document.content} />
+    </div>
   )
 }
 
-export default Post
+export default Homepage
+```
 
-export const pageQuery = graphql`
-  query PostBySlug($uid: String!) {
-    prismicPost(uid: { eq: $uid }) {
-      uid
-      data {
-        title {
-          text
-        }
-        content {
-          html
+### Repeatable Type
+
+Because you defined a custom type as `Post` the query after the `prismic` key is called `allPosts` (and `Post`). You can also see the API IDs (from the field names) you created earlier.
+
+```js:title=src/templates/post.js
+import React from "react"
+import { graphql } from "gatsby"
+import { RichText } from "prismic-reactjs"
+
+export const query = graphql`
+  query PostQuery($uid: String) {
+    prismic {
+      allPosts(uid: $uid){
+        edges {
+          node {
+            _meta {
+              uid
+            }
+            title
+            content
+          }
         }
       }
     }
   }
 `
+const Post = props => {
+  const prismicContent = props.data.prismic.allPosts.edges[0]
+  if (!prismicContent) return null
+
+  const document = prismicContent.node
+
+  return (
+    <div>
+      <RichText render={document.title} />
+      <RichText render={document.content} />
+    </div>
+  )
+}
+
+export default Post
 ```
```

Comment:
This needs a section about running the `gatsby develop` command, so people can see what they created before deploying. 

Unfortunately, I'm running into a bug with the `gatsby-source-prismic-graphql` plugin about ` Error: Cannot find module './third-party/gatsby-node'`. It would help to have someone else run through this guide to see if you're also affected by it. I'd hesitate to publish a guide for a source plugin that doesn't work. https://github.com/birkir/gatsby-source-prismic-graphql/issues/217

Reply:
Ok. I had a paragraph where it explains that they can run gatsby develop to test in dev, and to test queries on line 87, now I've added it on line 173 as well as you're suggesting.

And yes, this is a known issue we are aware of. And the reason we are reconsidering moving the docs, as I also note at the end of this review.